.github/workflows/integ-tests.yml

```diff
@@ -0,0 +1,211 @@
+name: Integ Tests
+
+# TODO set this up to gate new releases of Syne Tune
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+  pull_request: # remove before merging, this is for testing
+    branches:   # remove before merging, this is for testing
+      - main    # remove before merging, this is for testing
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read  # This is required for actions/checkout
+
+env:
+  AWS_DEFAULT_REGION: us-west-2
+  AWS_ROLE: ${{ secrets.PROD_AWS_INTEG_TEST_ROLE_ARN }}
+
+jobs:
+  stop_all_training_jobs: 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PROD_AWS_INTEG_TEST_ROLE_ARN }}
+          role-session-name: integtestsession
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+      - name: Stop all left-over training jobs
+        run: |
+          aws sagemaker list-training-jobs --status-equals InProgress > running_jobs.json
+          jq -c '.[][]["TrainingJobName"]' running_jobs.json | while read i; do
+            jobName=`echo $i | cut -d "\"" -f 2`
+            echo "stopping training job $jobName"
+            aws sagemaker stop-training-job --training-job-name $jobName
+            sleep 5
+          done
+
+  launch_tensorboard_example:
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      script-path: examples/launch_tensorboard_example.py
+    secrets: inherit
+  
+  launch_tuning_gluonts:
+    needs: stop_all_training_jobs
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      script-path: examples/launch_tuning_gluonts.py
+      requires-aws-credentials: true
+    secrets: inherit
+
+  launch_plot_results:
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      additional-command: pip install matplotlib
+      script-path: examples/launch_plot_results.py
+    secrets: inherit
+
+  launch_rl_tuning:
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      additional-command: pip install -r examples/training_scripts/rl_cartpole/requirements.txt
+      script-path: examples/launch_rl_tuning.py
+    secrets: inherit
+
+  launch_simulated_benchmark:
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      script-path: examples/launch_simulated_benchmark.py
+    secrets: inherit
+
+  launch_standalone_bayesian_optimization:
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      script-path: examples/launch_standalone_bayesian_optimization.py
+    secrets: inherit
+
+  launch_moasha_instance_tuning:
+    needs: stop_all_training_jobs
+    uses: ./.github/workflows/run-syne-tune.yml
+    with:
+      script-path: examples/launch_moasha_instance_tuning.py
+      requires-aws-credentials: true
+    secrets: inherit
+
+  launch_nas201_transfer_learning:
+    uses: ./.github/workflows/run-syne-tune.yml
```

Comment:
This is fine with me as is, but why not always adding `needs: stop_all_training_jobs`?
It does cost if no job needs to be stopped but it makes it allow to not forget to add this line (which can happen if you think you dont need it whereas the examples actually spawed SM jobs).

Reply:
Good point, there's some duplication of tests between this file and the other one. I think we could do the following: 
1. Rename this file to be "End-to-end tests"
2. Remove all tests that are in the "Fast tests" from this file
3. All remaining tests in this file will always need the `needs: stop_all_training_jobs` line in this file

This way the role of each test suite is more clear

See this earlier comment also for more of what I'm thinking now, based on your comment: https://github.com/awslabs/syne-tune/pull/520#discussion_r1095523249